stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/filepreview/AttachmentPreviewHandler.kt

```diff
@@ -0,0 +1,41 @@
+package io.getstream.chat.android.compose.ui.filepreview
+
+import android.content.Context
+import io.getstream.chat.android.client.models.Attachment
+
+/**
+ * Represents a handler that can show a preview for an attachment of certain type.
+ */
+public interface AttachmentPreviewHandler {
+
+    /**
+     * Determines if the current preview handler is responsible for showing preview for the given attachment.
+     *
+     * @param attachment The attachment we want to show the preview for.
+     * @return True if the current handler can provide a preview for this attachment.
+     */
+    public fun canHandle(attachment: Attachment): Boolean
+
+    /**
+     * Shows a preview for this attachment using the current preview handler.
+     *
+     * @param attachment The attachment we want to show the preview for.
+     */
+    public fun handleAttachmentPreview(attachment: Attachment)
```

Comment:
Maybe pass the URL here instead? If we're only going to use that

Reply:
Not so sure. In the UI components there is a logic which shows audio logo for audio files. I don't show it as it covers the play/pause icon, but still we might want to do something similar in future. Also, would be nice to show file names for video/audio files. I would keep this generic.

Reply:
The new `MediaAttachmentActivity` contains a header with the attachment's title. You can check latest UI components app for reference